### ISSUE TITLE
Fuzzing webm/libvpx

### ISSUE THREAD

**Issue by Splines** [aflgo/aflgo#100]
Hi there,

I'm trying to fuzz [**libvpx**](https://www.webmproject.org/code/) - a video codec library from Google - using AFLGo. However, linking fails at the moment.

## My setup

```sh
export CC=$AFLGO/afl-clang-fast
export CXX=$AFLGO/afl-clang-fast++

export ADDITIONAL="-targets=$TMP_DIR/BBtargets.txt -outdir=$TMP_DIR -flto -fuse-ld=gold -Wl,-plugin-opt=save-temps"
pushd $SUBJECT
    mkdir -p libvpx-aflgo-build
    cd ./libvpx-aflgo-build
    ../configure --extra-cflags="$ADDITIONAL" --disable-docs --disable-examples --log=<my-log-file-path>
    make
popd
```

**When I comment out the CC and CXX exports and remove the `--extra-cflags` I am able to successfully finish the build of libvpx.**

## Log
Some extracts from the log that might be interesting:

<details>
  <summary>error: "not x32"</summary>

  ```
Configuring for target 'x86_64-linux-gcc'
  enabling x86_64
check_cpp
BEGIN /tmp/vpx-conf-68549-27170.c
     1	#ifndef __ILP32__
     2	#error "not x32"
     3	#endif
END /tmp/vpx-conf-68549-27170.c
/home/ubuntu/build/llvm_tools/build-llvm/msan/aflgo/afl-clang-fast -E -o /tmp/vpx-conf-68549-27170.o /tmp/vpx-conf-68549-27170.c
/tmp/vpx-conf-68549-27170.c:2:2: error: "not x32"
#error "not x32"
 ^
1 error generated.
```
</details>

<details>
<summary>File format not recognized (this is how the log ends)</summary>

```
BEGIN /tmp/vpx-conf-68549-27170.c
     1	int main(void) {return 0;}
END /tmp/vpx-conf-68549-27170.c
/home/ubuntu/build/llvm_tools/build-llvm/msan/aflgo/afl-clang-fast -m64 -DNDEBUG -O3 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -targets=<my-path-to-bbtargets.txt> -outdir=<my-path-to-out-folder> -flto -fuse-ld=gold -Wl,-plugin-opt=save-temps -Wall -Wdeclaration-after-statement -Wdisabled-optimization -Wpointer-arith -Wtype-limits -Wcast-qual -Wvla -Wimplicit-function-declaration -Wuninitialized -Wunused-variable -fno-strict-aliasing -Wno-unused-function -c -o /tmp/vpx-conf-68549-27170.o /tmp/vpx-conf-68549-27170.c
/home/ubuntu/build/llvm_tools/build-llvm/msan/aflgo/afl-clang-fast++ -m64 -o /tmp/vpx-conf-68549-27170.x /tmp/vpx-conf-68549-27170.o -lpthread
/tmp/vpx-conf-68549-27170.o: file not recognized: File format not recognized
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```
</details>

Any help is much appreciated 😇

**Comment by strongcourage** [comment]
@Splines maybe it may not work as expected in next steps if you remove this compilation param :)

**Comment by mboehme** [comment]
We need the Gold plugin in the first stage to generate the CG and CFGs. We recently upgraded to LLVM 11.0. Do you mean Gold is not needed for the second stage or do you mean Gold is not needed for the CFG/CG generation?

**Comment by Splines** [comment]
@mboehme I mean for the CFG/CG generation. I was able to extract the dot files even without the flags `-flto -fuse-ld=gold`. With those flags, I can't even run `configure` successfully as it ends with: "Toolchain is unable to link executables".

Without the Gold plugin I can run `configure` and successfully compile (`aflg-compiler (yeah!)`messages in the terminal). However, I do get linking erros like this one:

```
strip: stQfQfCz/vp9_reconintra.c.o: Failed to find link section for section 19
```

And therefore, the build directory does not contain any binaries :(

**Comment by mboehme** [comment]
@strongcourage: If you have a few spare cycles, can you please check if Gold is needed for the programs in your tutorials?

@Splines: so ultimately you still get linker errors? Can you build without AFLGo and with LLVM 11?

**Comment by Splines** [comment]
@mboehme Yes, ultimately I still get linker error. I was able to build it without AFLGo simply by using the default configuration for `configure` as seen in the [libvpx build instructions](https://chromium.googlesource.com/webm/libvpx.git/+/refs/heads/main/README). But I don't know how to figure out which LLVM version was used then. That's why I tried to explicitly install LLVM 11 and use the clang binary from there.

I went [here](https://apt.llvm.org/) to see the instructions on how to download LLVM 11. So I ran on Ubuntu 18.04:

```
wget https://apt.llvm.org/llvm.sh
chmod +x llvm.sh
sudo ./llvm.sh 11
```

However, now I get the following error. I couldn't figure out to solve it. The comments in [this](https://stackoverflow.com/q/54273632/9655481) stack overflow post suggest to add a `target_link_libraries(codegen LLVM)` to the CMake file, however libvpx doesn't even have a CMake file...

```
: CommandLine Error: Option 'help-list' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
```
